style-dictionary/templates/scss-color-map.template

```diff
@@ -0,0 +1,58 @@
+<%
+//
+// Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// You may not use this file except in compliance with the License.
+// A copy of the License is located at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+%>
+<%
+    // Modified version of map-deep.template provided by style-dictionary.
+    //
+    // output the list of tokens as a Sass nested map
+    // (the values are pointing to the variables)
+    //
+    // Numbers keys will be output without quotes and paired with their value.
+    // Strings values with a nested maps will be output with quotes and their nested maps will be
+    // evaluated.
+    //
+    print(`$${file.mapName||'tokens'}: ${processJsonNode(dictionary.properties, 0)};\n`);
+
+    // recursive function to process a properties JSON node
+    //
+    function processJsonNode(obj, depth) {
+        let output = '';
+        if (obj === null) {
+            output += `''`
+        } else if (typeof obj === "string" || obj === false) {
+            // Allows the output of `false` values
+            output += `${obj}`
+        } else if (obj.hasOwnProperty('value')) {
+            // if we have found a leaf (a property with a value) append the value
+            output += `${obj.name}`;
+        } else {
+            // if we have found a group of properties, use the Sass group "(...)" syntax and loop -recursively- on the children
+            output += '(\n'
+            output += Object.keys(obj).map(function(newKey) {
+            let newProp = obj[newKey];
+            let indent = '  '.repeat(depth+1);
+            // if newKey is a number, do not wrap with quotation marks, process newProp value
+            // else wrap with quotation marks and process nested newProp object.
+            if (!isNaN(Number(newKey))){
+              return `${indent}${newKey}: ${processJsonNode(newProp.value, depth + 1)}`;
+            } else {
```

Comment:
This section of the transformer was added by me to ensure number keys did not get wrapped in quotation marks. This was necessary to match the current token structure.

Strings are wrapped in quotation marks, numbers are not.